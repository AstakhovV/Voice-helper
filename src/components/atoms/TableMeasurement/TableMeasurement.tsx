import React from 'react';
import { usePower } from '../../../application/usePower';

const TableMeasurement = () => {
  const { realMeasurement } = usePower();

  return (
    <table className="table-auto w-80 text-black border-2">
      <thead>
        <tr>
          <th />
          <th className="border-2">Line 1</th>
          <th className="border-2">Line 2</th>
        </tr>
      </thead>
      <tbody>
        {realMeasurement.map((el, i) => (
          <tr key={i}>
            <td className="border-2 font-bold">{el.name}</td>
            <td className="border-2">{`${el.value1} ${el.unit}`}</td>
            <td className="border-2">{`${el.value2} ${el.unit}`}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default TableMeasurement;
