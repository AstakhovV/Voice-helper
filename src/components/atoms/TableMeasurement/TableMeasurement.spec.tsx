import TableMeasurement from './TableMeasurement';
import React from 'react';
import { act, render, screen } from '@testing-library/react';
import PowerStationProvider from '../../../application/usePower';

describe('Render TableMeasurement', () => {
  test('TableMeasurement renders correctly', () => {
    render(
      <PowerStationProvider>
        <TableMeasurement />
      </PowerStationProvider>,
    );
    const table = screen.getByRole('table');
    expect(table).toMatchSnapshot();
  });

  test('TableMeasurement has 3x5 grid', () => {
    render(
      <PowerStationProvider>
        <TableMeasurement />
      </PowerStationProvider>,
    );
    const table = screen.getByRole('table');
    const row = table.querySelectorAll('tr');
    const col = table.querySelectorAll('th');
    expect(row).toHaveLength(5);
    expect(col).toHaveLength(3);
  });

  test('TableMeasurement has changing values', async () => {
    jest.useFakeTimers();

    render(
      <PowerStationProvider>
        <TableMeasurement />
      </PowerStationProvider>,
    );
    const tableItemVoltage = screen.getAllByRole('cell')[1];
    const initialValue = '10.4 kV';
    // @ts-ignore
    act(() => jest.advanceTimersByTime(1000));
    expect(tableItemVoltage).not.toHaveTextContent(initialValue);
  });
});
