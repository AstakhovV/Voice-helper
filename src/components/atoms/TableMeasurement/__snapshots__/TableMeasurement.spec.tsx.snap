// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TableMeasurement TableMeasurement renders correctly 1`] = `
<table
  class="table-auto w-80 text-black border-2"
  role="table"
>
  <thead>
    <tr>
      <th />
      <th
        class="border-2"
      >
        Line 1
      </th>
      <th
        class="border-2"
      >
        Line 2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="border-2 font-bold"
      >
        Voltage
      </td>
      <td
        class="border-2"
        role="cell"
      >
        10.4 kV
      </td>
      <td
        class="border-2"
      >
        10.3 kV
      </td>
    </tr>
    <tr>
      <td
        class="border-2 font-bold"
      >
        Current load
      </td>
      <td
        class="border-2"
        role="cell"
      >
        53 A
      </td>
      <td
        class="border-2"
      >
        46 A
      </td>
    </tr>
    <tr>
      <td
        class="border-2 font-bold"
      >
        Power
      </td>
      <td
        class="border-2"
        role="cell"
      >
        551.2 kVA
      </td>
      <td
        class="border-2"
      >
        473.8 kVA
      </td>
    </tr>
    <tr>
      <td
        class="border-2 font-bold"
      >
        Frequency
      </td>
      <td
        class="border-2"
        role="cell"
      >
        50 Hz
      </td>
      <td
        class="border-2"
      >
        49.9 Hz
      </td>
    </tr>
  </tbody>
</table>
`;
